shell/envvar_cleanup.bash

```diff
@@ -0,0 +1,327 @@
+#!/usr/bin/env bash
```

Comment:
This comment shouldn't block merging this PR, but...  I'm pausing and thinking a little bit here.  This effectively makes bash a hard-dependency of conda.  (Not that it wasn't before, but this PR is all about cleaning things up!)

It's not jumping out at me right now, but I'm guessing you use some bash-ness somewhere that's not strictly posix?

I doubt anybody will care, but this could come up as an issue.  And if it does, it seems to me there are several possibilities on how to proceed.  The most obvious one to me would be to write a separate entry_point or `conda ..XXXXX` helper command.  Obviously that returns *quickly* and doesn't load a ton of conda code imports.  Then we could also unify the duplicated logic that's in the .bat version.